pydantic_ai_slim/pydantic_ai/models/anthropic.py

```
@@ -2960,6 +3013,43 @@ def _map_mcp_server_result_block(
+
+    Code execution containers are short-lived, so a download can fail (expired container, rate limit, transient
+    error). In that case we skip the file with a warning rather than discarding the otherwise-successful response.
+    """
+    # The Files API is only available on the direct Anthropic client, not the Bedrock/Vertex variants.
+    client = cast(AsyncAnthropic, client)
+    try:
+        file_content = await client.beta.files.download(file_id=file_id)
+        file_bytes = await file_content.read()
+        metadata = await client.beta.files.retrieve_metadata(file_id=file_id)
+    except (APIStatusError, APIConnectionError) as e:
+        warnings.warn(f'Failed to download code execution file {file_id}: {e}', UserWarning)
```

Ruff (0.15.20)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)